sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -177,6 +177,15 @@ object DataSourceV2Strategy extends Strategy with PredicateHelper {
           throw new AnalysisException("DELETE is only supported with v2 tables.")
       }
 
+    case VacuumTable(relation) =>
+      relation match {
+        case DataSourceV2ScanRelation(table, _, output) =>
+          VacuumTableExec(table.asVacuumable) :: Nil
+        case _ =>
+          throw new AnalysisException("VACUUM is only supported with v2 tables.")
```

Comment:
This isn't the right place to throw an `AnalysisException` because this is after analysis has finished.

This rule should match `VacuumTable(DataSourceV2ScanRelation(table, _, _))` and produce the exec node. Then `CheckAnalysis` should have a rule that matches `VacuumTable` with any other relation that throws the analysis exception.